### DISCUSSION TITLE
OAP init doubt

### DISCUSSION THREAD

**Discussion by larrying** [apache/skywalking#9253] [Q&A] [answered]
I have some doubts about OAP initialization. I looked at the source code. If the mode env is not set to "no-init", the initialization actions(es templates/indices) will be executed.
In other words, is the 'oap-init-job' in the show case necessary?

https://github.com/apache/skywalking-showcase/blob/main/deploy/platform/kubernetes/feature-cluster/resources.yaml

**Comment by wu-sheng** [comment, marked as answer]
Yes, it is necessary. ElasticSearch server could be easily in deadlocked if multiple OAP service pods start concurrently.

**Reply by larrying** [reply]
Understood, I used to start a pod first, so there was no problem when initializing

**Reply by wu-sheng** [reply]
That is what the init container did. Manual operation V.S. k8s natively.
